### PR TITLE
Add :db.type/{instant, uri, uuid} types to bootstrapper. Fixes #201

### LINKED ISSUE #201: [db] [tx] Support :db.type/{instant,uuid,uri}
Over in #170, I simplified things by not including placeholders for Mentat value types I wasn't going to support at the start: `:db.type/instant`, `:db.type/uuid`, and `:db.type/uri`.  (We might consider supporting JSON and a binary blob type like `:db.type/bytes` as well.)

This ticket tracks supporting these types.  Off the top, this will mean:
1. Adding new `DB_TYPE_*` values to the bootstrapper;
1. Bumping the SQL schema to accommodate the new idents;
1. Adding new `ValueType` cases;
1. Adding corresponding `TypedValue` cases;
1. Implementing the conversions to and from SQL;
1. testing the new types in the transactor and potentially in the query engine as well.

### PR DESCRIPTION
**Pull request by jsantell**
This is missing a v3.db to run in the `test_open_current_version` test in `db/src/db.rs`,but looking for a sanity check here. Commenting out that test, everything else passes. Unlikely I'll tackle the creation of a v3 DB in `./fixtures` and the SQL update migration for v1 -> v2 -> v3

### CODE REVIEW

File: db/src/bootstrap.rs [hunk 1]
```diff
@@ -138,6 +152,16 @@ lazy_static! {
             .ok_or(ErrorKind::BadBootstrapDefinition("Unable to parse V2_SYMBOLIC_SCHEMA".into()))
             .unwrap()
     };
+
+    static ref V3_SYMBOLIC_SCHEMA: Value = {
+        let s = r#"{}"#;
```
Comment (jsantell) [inline db/src/bootstrap.rs]: This doesn't do anything, because the symbolic schema doesn't change, AFAIK?

File: db/src/bootstrap.rs [hunk 2]
```diff
@@ -233,27 +257,27 @@ fn symbolic_schema_to_assertions(symbolic_schema: &Value) -> Result<Vec<Value>>
 }
 
 pub fn bootstrap_partition_map() -> PartitionMap {
-    V2_PARTS[..].iter()
+    V3_PARTS[..].iter()
```
Comment (jsantell) [inline db/src/bootstrap.rs]: I think changing all these variables to the latest is annoying. Should just have a struct for each schema that contains parts, idents, etc., and just reference latest via `const LATEST = V3_Schema` and use that throughout.

File: db/src/bootstrap.rs (hunk 1 above)
Comment (ncalexan) [inline db/src/bootstrap.rs]: Correct, and you're welcome to just clone the `V2_SYMBOLIC_SCHEMA` or whatever is needed if you prefer.

The symbolic schema would/will change if/when we make `:db/txInstant` have `:db.valueType :db.type/inst`, like Datomic.

File: db/src/bootstrap.rs (hunk 2 above)
Comment (ncalexan) [inline db/src/bootstrap.rs]: I'm fine with this.  Do it!  Or good first bug it and mentor it.

**Comment by jsantell** [comment]
Rebased, but still failing in the test that uses the v2 database @ `db/src/db.rs:884`

**Comment by rnewman** [comment]
Obsoleted by #438.

**Pull request closed without merging**